dandi/models.py

```diff
@@ -823,6 +880,17 @@ class PublishedDandisetMeta(DandisetMeta):
         nskey="dandi",
     )  # TODO: formalize "publish" activity to at least the Actor
     datePublished: date = Field(readOnly=True, nskey="schema")
```

Comment:
note: we need to streamline (and probably adjust descriptions here here) the life cycle of all the `date*` markers, as relating to how `dandi-api` would treat/tune them.
ATM it is not immediately clear to me how this one would be different e.g. from `dateModified` since we should not modify published dandisets:
- I guess if `dateModified` would be "filled in" by `dandi-api` and reflect the date for the database row entry, and we adjust any columns which are not in the `metadata` (stats?) that would lead to `dateModified` change
- but still not clear how it would be different from `dateCreated`?  If dateCreated would be date of creation of dandiset (not filled in from DB row table), then `datePublished` would have purpose.  But if `dateCreated` is for the date of when published row record was created -- they should/would be identical. 

Meanwhile:
```suggestion
    datePublished: date = Field(readOnly=True,  title="Publication date and time", nskey="schema")
```

Reply:
datePublished could be 5 days after last modified, unless we say the publishing action is a modification action on the dandiset. this really becomes a notion of what we expect modified to mean.

all dates are filled in from database table.